RecoLocalCalo/CaloTowersCreator/src/CaloTowersCreationAlgo.cc

```diff
@@ -1629,11 +1640,34 @@ void CaloTowersCreationAlgo::makeHcalDropChMap() {
 	bool merge = theHcalTopology->mergedDepth29(hid);
 	if (merge) {
           CaloTowerDetId twrId29(twrId.ieta()+twrId.zside(), twrId.iphi());
-          hcalDropChMap[twrId29] +=1;
+          hcalDropChMap[twrId29].first +=1;
 	}
       }
     }
-
+  }
+  // now I know how many bad channels, but I also need to know if there's any good ones
+  if (missingHcalRescaleFactorForEcal > 0) {
+      for (auto & pair : hcalDropChMap) {
+          if (pair.second.first == 0) continue; // unexpected, but just in case
+          int ngood = 0, nbad = 0;
+          for (DetId id : theTowerConstituentsMap->constituentsOf(pair.first)) {
+              if (id.det() != DetId::Hcal) continue;
+              HcalDetId hid(id);
+              if (hid.subdet() != HcalBarrel && hid.subdet() != HcalEndcap) continue;
+              const uint32_t dbStatusFlag = theHcalChStatus->getValues(id)->getValue();
+              if (dbStatusFlag == 0 || ! theHcalSevLvlComputer->dropChannel(dbStatusFlag)) {
+                  ngood += 1;
+              } else {
+                  nbad += 1; // recount, since pair.second.first may include HO
+              }
+          }
+          if (nbad > 0 && nbad >= ngood) {
+              CaloTowerDetId id(pair.first);
```

Comment:
this better be placed in the same commented region in the lines below (or placed in some if debug)